### PR TITLE
Hide Obsolete warnings on record initializer not using the obsolete field

### PR DESCRIPTION
**Pull request by edgarfgp**
Attempt to Fix https://github.com/dotnet/fsharp/issues/14203

### CODE REVIEW

**Comment by edgarfgp** [comment]
Should this also behind a preview flag ?

**Comment by dsyme** [comment]
I'm OK with considering this a bug fix,

**Comment by edgarfgp** [comment]
@fsharp-team-msft I'm a little stuck here. Would appreciate some pointers or if you are willing to take this over will be also fine by me. I do not want to hold this from being fixed.

File: src/Compiler/Checking/CheckExpressions.fs [hunk 1]
```diff
@@ -4616,7 +4616,7 @@ and TcMeasuresAsTuple (cenv: cenv) newOk checkConstraints occ env (tpenv: Unscop
         | SynTupleTypeSegment.Slash _ :: SynTupleTypeSegment.Type ty :: args ->
             let ms1, tpenv = TcMeasure cenv newOk checkConstraints occ env tpenv ty m
             gather args tpenv (Measure.Prod(acc, Measure.Inv ms1))
-        | _ -> failwith "inpossible"
```
Comment (vzarytovskii) [inline src/Compiler/Checking/CheckExpressions.fs:4619]: Aw, I liked the "inpossible" more!

**Comment by T-Gro** [comment]
In the code added by this PR, BuildFieldMap tries to bring fields from inner expressions to the outer type in case of a double-nested record update.
I added a failing test, this is where is produces on the nested with clause.
It passes the field "ccont" to OuterType, which does not match.

image

**Comment by edgarfgp** [comment]
Thanks @T-Gro  for the pointers. I think I have a better fix in https://github.com/dotnet/fsharp/pull/15705. Closing this

**Pull request closed without merging**